src/components/KEmptyState/KEmptyState.vue

```diff
@@ -1,143 +1,167 @@
 <template>
-  <section
-    class="empty-state-wrapper"
-    :class="{ 'is-error': isError }"
+  <div
+    class="k-empty-state"
+    :class="[iconVariant]"
   >
-    <div class="empty-state-title">
-      <div
-        v-if="isError || icon"
-        class="k-empty-state-icon card-icon"
-        :class="{ 'warning-icon': isError }"
-      >
-        <KIcon
-          :color="isError ? iconColor || `var(--kui-color-text-neutral-strongest, ${KUI_COLOR_TEXT_NEUTRAL_STRONGEST})` : iconColor"
-          :icon="icon ? icon : 'warning'"
-          :secondary-color="isError ? iconSecondaryColor || 'currentColor' : iconSecondaryColor"
-          :size="iconSize"
-        />
-      </div>
-      <div
-        v-if="$slots.title"
-        class="k-empty-state-title-header"
-      >
-        <slot name="title" />
-      </div>
-    </div>
     <div class="empty-state-content">
-      <div
-        v-if="$slots.message"
-        class="k-empty-state-message"
-      >
-        <slot name="message" />
+      <div class="empty-state-icon">
+        <slot name="icon">
+          <component
+            :is="getEmptyStateIcon"
+            :color="getIconColor"
+            :size="KUI_ICON_SIZE_60"
+          />
+        </slot>
       </div>
+      <span
+        v-if="title"
+        class="empty-state-title"
+      >
+        {{ title }}
+      </span>
       <div
-        v-if="$slots.cta || (!ctaIsHidden && ctaText)"
-        class="k-empty-state-cta"
+        v-if="message || $slots.default"
+        class="empty-state-message"
       >
-        <slot name="cta">
-          <KButton
-            appearance="primary"
-            size="small"
-            @click.prevent="() => handleClick && handleClick()"
-          >
-            {{ ctaText }}
-          </KButton>
+        <slot name="default">
+          <p>
+            {{ message }}
+          </p>
         </slot>
       </div>
     </div>
-  </section>
+    <div
+      v-if="actionButtonVisible && (actionButtonText || $slots.action)"
```

Comment:
**Issue**: This is a regression preventing the use of the action slot while `props.actionButtonVisible` is `false`. If `action` content is slotted in, no further condition should prevent the slot from rendering. If `action` content is not slotted in and both `props.actionButtonVisible` and `props.actionButtonText` are set, then the slot should render, too.

```suggestion
      v-if="$slots.action || (actionButtonVisible && actionButtonText)"
```

Seeing that `props.actionButtonVisible` defaults to true and that `props.actionButtonText` defaults to `''` (i.e. it’s required for users to provide the text if they want to rely on the action slot’s fallback content), the visibility prop is superfluous and can be removed, simplifying the check to:

```suggestion
      v-if="$slots.action || actionButtonText"
```

Reply:
We can workaround this by dynamically setting `actionButtonVisible` to `true` if we’re using `action` slot content and, technically, since `actionButtonText` isn’t set initially, we could leave `actionButtonVisible` unset where we sometimes want an action button here, but I think the second suggestion above makes using this a little simpler because it allows removing one prop.